src/__tests__/initRules.test.js

```diff
@@ -13,639 +13,2329 @@
+              generateCondition('Condition2')
+            ]
+          }
+        ]);
+
+        runInitRules(rules);
+
+        var conditionExport1 = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Condition1')
+        );
+        var conditionExport2 = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Condition2')
+        );
+
+        expect(conditionExport1.calls.count()).toBe(1);
+        expect(conditionExport2.calls.count()).toBe(0);
       });
 
-      var condition2Exports = moduleProvider.getModuleExports(TEST_CONDITION2_PATH);
+      it('does not execute actions when condition fails', function() {
+        var rules = setupRules([
+          {
+            conditions: [
+              generateCondition('Condition1', function(module) {
+                module.exports = jasmine.createSpy().and.returnValue(false);
+              })
+            ],
+            actions: [generateAction('Action1')]
+          }
+        ]);
 
-      expect(condition2Exports.calls.count()).toBe(1);
+        runInitRules(rules);
 
-      var condition2ExportsCall = condition2Exports.calls.mostRecent();
+        var conditionExport = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Condition1')
+        );
+        var actionExport = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Action1')
+        );
 
-      expect(condition2ExportsCall.args[0]).toEqual({
-        testCondition2Foo: 'bar'
+        expect(conditionExport.calls.count()).toBe(1);
+        expect(actionExport.calls.count()).toBe(0);
       });
 
-      expect(condition2ExportsCall.args[1]).toEqual({
-        $type: 'test-extension.test-event',
-        $rule: {
-          id: 'RL123',
-          name: 'Test Rule'
-        }
+      it('does not execute actions when negated condition fails', function() {
+        var rules = setupRules([
+          {
+            conditions: [
+              generateNegatedCondition('Condition1', function(module) {
+                module.exports = jasmine.createSpy().and.returnValue(true);
+              })
+            ],
+            actions: [generateAction('Action1')]
+          }
+        ]);
+
+        runInitRules(rules);
+
+        var conditionExport = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Condition1')
+        );
+        var actionExport = moduleProvider.getModuleExports(
+          ModuleHelper.getPath('Action1')
+        );
+
+        expect(conditionExport.calls.count()).toBe(1);
+        expect(actionExport.calls.count()).toBe(0);
       });
 
-      var action1Exports = moduleProvider.getModuleExports(TEST_ACTION1_PATH);
+      it('executes the events then the conditions then the actions', function() {
+        var callOrder = [];
+        var rules = setupRules([
+          {
+            events: [
+              generateEvent('Event1', function(module) {
+                callOrder.push('Event1');
+                module.exports = jasmine
```

Comment:
I prefer we just use regular functions unless we're asserting against the spies.

Reply:
Done.